src/sentry/plugins/sentry_mail/models.py

```diff
@@ -317,11 +323,45 @@ def notify_digest(self, project, digest):
         headers = {
             'X-Sentry-Project': project.slug,
         }
-
         group = six.next(iter(counts))
         subject = self.get_digest_subject(group, counts, start)
 
+        if ProjectOwnership.objects.filter(project_id=project.id).exists():
+            self.send_digest_with_owners(context, project, subject, headers)
+            return
+
+        for user_id in self.get_send_to(project):
+            self.add_unsubscribe_link(context, user_id, project)
+            self._send_mail(
+                subject=subject,
+                template='sentry/emails/digests/body.txt',
+                html_template='sentry/emails/digests/body.html',
+                project=project,
+                reference=project,
+                headers=headers,
+                type='notify.digest',
+                context=context,
+                send_to=[user_id],
+            )
+
+    def send_digest_with_owners(self, context, project, subject, headers):
+        original_context = context
+        events = get_events_from_digest(context['digest'])
+        event_actors = ProjectOwnership.get_all_actors(project.id, [event[0] for event in events])
+        event_users = self.event_actors_to_user_ids(event_actors)
+
         for user_id in self.get_send_to(project):
+            context = original_context
+            events_for_user = [event for event in events if user_id in event_users[event[0]]]
+            if not events_for_user:
+                continue
+            if len(events_for_user) != len(original_context['counts']):
```

Comment:
Ah... that's a good catch. It will in fact. I'm not sure how best to fix this. What I'm trying to say is if we've not filtered out any events. I could use a boolean value perhaps to signify that an event has been filtered?

Reply:
The main thing to keep in mind here (I didn't do a great job articulating this above) is that [`counts` is a mapping of groups to the number of rules that they were associated with in the digest](https://github.com/getsentry/sentry/blob/1b498d59279e9b59faea59eb08ad736651717814/src/sentry/digests/utilities.py#L15) so just be extra careful that you're comparing similar things here.